hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieTimeline.java

```diff
@@ -232,6 +233,12 @@
    */
   Option<byte[]> getInstantDetails(HoodieInstant instant);
 
+  /**
+   * Returns partitions that have been modified in the timeline. This includes internal operations such as clean.
+   * Note that this only returns data for completed instants.
+   */
+  List<String> getPartitionsMutated();
```

Comment:
I don't think this makes sense in the timeline as of now. If you take a look at the timeline API's,  they only talk about the metadata that has changed. `getPartitionsMutated` conceptually is providing what has changed in the underlying data as opposed to what has changed in the timeline per se. Generally, all of this information should come from the timeline but that requires a full redesign on the timeline. Should we add this API here -> https://github.com/apache/hudi/blob/master/hudi-client/src/main/java/org/apache/hudi/client/HoodieReadClient.java#L195 ? And you can wrap this functionality in a TimelineUtils ? 
When we have clearer design on timeline, we can merge back the TimelineUtils to the real timeline...

Reply:
+1 on abstraction point. I think having a separate helper class would be better.

Reply:
@n3nash I want this to be in hudi-common, so this can be reused in hadoop-mr and hive-sync.  Do you want me to create TimelineUtils in common?

Reply:
created TimelineUtils